android-components/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt

```diff
@@ -183,4 +187,21 @@ abstract class PlacesStorage(
             SyncStatus.Error(e)
         }
     }
+
+    /**
+     * Registers a storage maintenance worker that prunes database when its size exceeds a size limit.
+     * */
+    override fun registerStorageMaintenanceWorker() {
+        // See child classes for implementation details, it is not implemented by default
+    }
+
+    /**
+     * Unregisters the storage maintenance worker that is registered
+     * by [PlacesStorage.registerStorageMaintenanceWorker].
+     *
+     * @param uniqueWorkName Unique name of the work request that needs to be unregistered
+     * */
+    override fun unregisterStorageMaintenanceWorker(uniqueWorkName: String) {
+        // See child classes for implementation details, it is not implemented by default
+    }
```

Comment:
Why i did do this way? 
To ensure that all the children do not need to know about the interface (compilation safety) and make it flexible for any usecase which may include either, both or none of the implementations of these two methods. 
Ex: We have maintenance task only for `PlacesHistoryStorage` and we only have the registering logic. Therefore, only `registerStorageMaintenanceWorker` method is implemented in `PlacesHistoryStorage` class. 
Future Ref: In case `RemoteTabsStorage` is required to have a maintenance task, then it would only require to implement `StorageMaintenanceRegistry` and provide custom implementations for these methods.